### ISSUE TITLE
Unable to load the Core in my MigStorm

### ISSUE THREAD

**Issue by jjsaenz1969** [wiretap-retro/MigStorm-ITX#1]
Thank you very much for all your work, this project is fantastic.
I have asked JLCPCB to manufacture your board, the BOM I have ordered the one you recommend on GitHub. And the memory and FPGA are from AliExpress.

![20241221_192643](https://github.com/user-attachments/assets/68f654b9-82c4-40bf-80be-bc1d020981f3)

![20241228_234932](https://github.com/user-attachments/assets/a8b35131-1b82-4132-a8cf-322e19918b7d)

My problem is that with the Pic and Core code you have on GitHub I get a black screen.
Whereas if I try other combinations of files downloaded from other sites it starts to load the Core.
But it never loads the whole Core.

![20241227_183959](https://github.com/user-attachments/assets/5c1bea89-eadc-4569-8acb-c2932cea9f43)

To program the pic I use the TL866II programmer with this combination of fuses.

![20241228_231611](https://github.com/user-attachments/assets/24283f0a-34f6-4b93-a378-c88ada8efeed)

The file ‘firstflash_bootloader+KGL140914.hex’ you have on Github is empty, it would be nice if you can put the correct one.

Let's see if I can get this new Amiga working.

**Comment by wiretap-retro** [comment]
It depends on the PIC you're using. You're going to want to use the tinybld198 folder for generating the PIC's firmware to flash with the firmware.hex file. The KGL140914 is for the SuperPIC. I've updated the firmware folder to include all variants of the PIC/SuperPIC/ARM modules. The original PIC is the 18F252, the SuperPIC is the 18F26K20, and the ARM is the AT91SAM7S256.

**Comment by jjsaenz1969** [comment]
Thank you for your prompt response.
I have the SuperPIC, then I will try what you have posted on Github.
Hopefully we'll finish this year with one more Amiga.

**Comment by jjsaenz1969** [comment]
No luck, it seems that the tl866 programmer is not doing a good job here.
I have ordered boards to make the ARM module and also a Null Modem cable to try to program via serial.

**Comment by wiretap-retro** [comment]
I typically use a Pickit2 or Pickit3 for programming the Pic chips. I'll keep this issue open for another month or so to see if you need additional help.

**Comment by jjsaenz1969** [comment]
Thank you for your advice.
I had a hard time finding it, but I remembered I had one.
I've never used it before, it was easy.

![Image](https://github.com/user-attachments/assets/33ce4d59-2447-41d8-bfbf-334cc7b3175c)

I was able to burn the chip with the file ‘firstflash_bootloader+KGL140914.hex’, but the result is the same.
It loads the Core but does not start the Minimig.

![Image](https://github.com/user-attachments/assets/25a65478-2d19-4a40-9ecd-ac444f4a6862)

The PacMan Core works perfectly, so I have a fault in the RAM or in the CPU.

Next I'll remove the CPU socket and check the chips underneath.

**Comment by wiretap-retro** [comment]
Yeah you'll want to check all orientations of the chips, and use a small sewing needle to wiggle each leg of each chip to make sure they are all soldered down right. Also check for bridges between legs. Sometimes they're really hard to spot without good magnification.

**Comment by jjsaenz1969** [comment]
I have good news.

I have checked the chips that are under the CPU, for this I created a new interactive BOM, but indicating that it shows the pads and tracks.
All correct there.
I also bought new RAM, this is from Mouser, the one I had before I bought from Aliexpress.

The unsoldered chip is Mouser's.
![Image](https://github.com/user-attachments/assets/2274813f-022c-47cf-ad34-26525cbdf8a6)

I have also checked every pin on the FPGA and everything is OK.
It's always the same, it always freezes at the same point.

My PIStorm is not the normal one, it is an edition that was made in a Spanish forum, it is a bit displaced so as not to touch with the keyboard on an A500.
I couldn't put it directly on the MigStorm as it clashed with the PicoPSU.
Today I got a CPU relocator, I just put it in, plugged my PIStorm and it started working.

The ARM module also works.

![Image](https://github.com/user-attachments/assets/7a069963-61c2-4463-9458-c296ccf28061)

![Image](https://github.com/user-attachments/assets/cdb635e1-8755-41fa-a93c-a036c39ed399)

![Image](https://github.com/user-attachments/assets/bda68c2e-1d5e-4eec-88ad-a3b2bcc51205)

Everything works very well, I'm very happy.
But I don't know why my MigStorm only works if it has a PIStorm on it.
I don't have a different accelerator to test.

**Comment by wiretap-retro** [comment]
I'm so glad to see you got it working. That's very good news. Hopefully you enjoy it!
